Add PR template (#163)
MetaMask/metamask-module-template: .github/pull_request_template.md

<!--
Thanks for your contribution! Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:

* What is the current state of things and why does it need to change?
* What is the solution your changes offer and how does it work?

> Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:

Reviewer: My perspective: Public PRs should not include internal links that need privileged access - some of these often are so I don't feel good about effectively normalizing that. This template will also be used by external contributors, often as a first point of interaction.
```suggestion
Are there any issues or other links reviewers should consult to understand this pull request better? For instance:
```

Author: Providing sufficient explanation in the PR to understand a change _without_ any additional context should be a goal ideally. But I don't see the harm in adding additional context, even if access is restricted. Pretending the relevant resources don't exist will not help anyone.

Regardless, your suggested change here looks good to me. This was just copied from one of our other templates.

Reviewer: I think if it's public information, it should be made accessible for readers of the PR. Useful context from Slack can be copy-pasted preferred to make it searchable) or shared as a screenshot, for example.

If it's confidential, it should not be linked.

* Fixes #12345
* See: #67890
-->

## Examples

<!--
Are there any examples of this change being used in another repository?

When considering changes to the MetaMask module template, it's strongly preferred that the change be experimented with in another repository first. This gives reviewers a better sense of how the change works, making it less likely the change will need to be reverted or adjusted later.

> -->

Reviewer 2: Can we add a section for breaking changes too?

Author: Hmm. I considered this, but I wasn't sure how to characterize a "breaking change" for the template. Wouldn't any change be "breaking" in a sense?
